addons/web/static/src/views/fields/domain/domain_field.xml

```diff
@@ -5,45 +5,77 @@
         <div t-att-class="{ o_inline_mode: !props.editInDialog }">
             <t t-set="resModel" t-value="getResModel()"/>
             <t t-if="resModel">
-                <DomainSelector
-                    resModel="resModel"
-                    domain="getDomain()"
-                    readonly="props.readonly or props.editInDialog"
-                    update.bind="update"
-                    isDebugMode="!!env.debug"
-                    className="props.readonly ? 'o_read_mode' : 'o_edit_mode'"
-                />
-                <div class="o_field_domain_panel d-flex align-items-center gap-2 mt-1">
-                    <i class="oi oi-arrow-right" role="img" aria-label="Domain" title="Domain" />
-                    <t t-if="state.isValid === null and state.recordCount === null">
-                        <i class="fa fa-circle-o-notch fa-spin ms-2" role="img" aria-label="Loading" title="Loading" />
-                    </t>
-                    <t t-else="">
-                        <t t-if="state.isValid">
-                            <button class="btn btn-sm btn-link o_domain_show_selection_button" type="button" t-on-click.stop="onButtonClick">
-                                <t t-esc="state.recordCount" /> record(s)
-                            </button>
+                <t t-if="props.isFoldable and state.folded">
+                    <div class="d-flex align-items-center" t-on-click="() => state.folded = false">
+                        <i data-tooltip="Modify filter" class="fa fa-lg fa-caret-right pe-2"/>
+                        <div class="d-print-contents">
+                            <t t-if="!state.facets.length">
+                                <span>Match <strong>all records</strong></span>
+                            </t>
+                            <t t-foreach="state.facets" t-as="facet" t-key="facet_index">
+                                <div class="d-inline-flex align-items-stretch text-nowrap px-1 position-relative o_facet_with_domain"
```

Comment:
the class o_facet_with_domain is not necessary here I think.
Please, review the classes and be sure that only the necessary classes are set